sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -295,12 +299,12 @@ class Dataset[T] private[sql](
    */
   def groupBy[K : Encoder](func: T => K): GroupedDataset[K, T] = {
     val inputPlan = queryExecution.analyzed
-    val withGroupingKey = AppendColumn(func, inputPlan)
+    val withGroupingKey = AppendColumn(func, resolvedTEncoder, inputPlan)
```

Comment:
We pass the `inputPlan` to `AppendColumn`, so `AppendColumn` should be able to resolve the encoder itself. Seems we can also pass unresolved encoder here?